probe-rs/src/flashing/flasher.rs

```diff
@@ -146,6 +178,17 @@ impl Flasher {
         })
     }
 
+    /// Check if target architecture supports CRC32 verification
+    fn target_supports_crc32(target: &Target) -> bool {
```

Comment:
It's somewhat confusing why there are two different functions, seemingly doing the same thing. Which one of these is supposed to return true for the risc-v targets, for which you provide the prebuilt binary?

Reply:
Thanks I'll have to test again; I had tried one of my small wch-link riscv only probes and thought probe-rs didn't detect it at all.